gnome-initial-setup/gis-page-util.c

```diff
@@ -446,3 +446,68 @@ gis_page_util_show_factory_dialog (GisPage *page)
 
   g_free (version);
 }
+
+typedef struct _DemoButtonData {
+  GtkDialog *demo_dialog;
+  GisDriver *driver;
+} DemoButtonData;
+
+static DemoButtonData *
+demo_button_data_new (GtkDialog *dialog, GisDriver *driver)
+{
+  DemoButtonData *data = (DemoButtonData *) g_slice_new (DemoButtonData);
+  data->demo_dialog = dialog;
+  data->driver = g_object_ref (driver);
```

Comment:
Do you really need to take a reference to the driver here? It should live for the whole duration of the FBE anyway